packages/deep/src/index.ts

```diff
@@ -0,0 +1,68 @@
+import { $PROXY } from "solid-js";
+import { Store } from "solid-js/store";
+
+/**
+ * deepTrack - create a deep getter on the given object
+ * ```typescript
+ * export function deepTrack<T>(
+ *   store: Store<T>
+ * ): T;
+ * ```
+ * @param store reactive store dependency
+ * @returns same dependency, just traversed deeply to trigger effects on deeply nested properties. For example:
+ *
+ * ```typescript
+ * createEffect(
+ *     on(
+ *         deepTrack(store),
+ *         () => {
+ *            // this effect will run when any property of store changes
+ *         }
+ *     )
+ *
+ * );
+ * ```
+ */
+export function deepTrack<T>(store: Store<T>): T {
+  return deepTraverse(store);
+}
+
+/**
+ * deepTraverse - traverse an object
+ * @param obj an object to traverse
+ * @returns an object with all values traversed
+ * @private
+ * */
+function deepTraverse<T>(value: Store<T>): T {
+  if (!isWrappable(value)) {
+    return value;
+  }
+
+  if (Array.isArray(value)) {
+    for (let i = 0; i < value.length; i++) {
+      deepTraverse(value[i]);
+    }
+  } else if (isWrappable(value)) {
```

Comment:
Is it necessary to check if the value is wrappable twice?
Also `isWrappable` also does `Array.isArray` check, so it could be removed either from here or from `isWrappable`.

Reply:
Oops, yeah, sorry, It should be good now!